static/app/components/forms/fields/sentryMemberTeamSelectorField.tsx

```diff
@@ -77,11 +85,36 @@ function SentryMemberTeamSelectorField({
     loadMore: loadMoreTeams,
   } = useTeams({provideUserTeams: true});
 
-  const teamOptions = teams?.map(team => ({
-    value: `team:${team.id}`,
-    label: `#${team.slug}`,
-    leadingItems: <Avatar team={team} size={avatarSize} />,
-  }));
+  // TODO(davidenwang): Fix the team type here to avoid this type cast
+  const teamOptions = (teams as DetailedTeam[])
+    .map(team => {
+      const isDisabledTeam =
+        memberOfProjectSlug &&
+        !defined(team.projects.find(({slug}) => memberOfProjectSlug === slug));
+      return {
+        value: `team:${team.id}`,
+        leadingItems: <Avatar team={team} size={avatarSize} />,
+        ...(isDisabledTeam
+          ? {
+              disabled: true,
+              label: (
+                <Tooltip
+                  position="left"
+                  title={t('%s is not a member of the selected project', `#${team.slug}`)}
+                >
+                  #{team.slug}
+                </Tooltip>
+              ),
+            }
+          : {
+              disabled: false,
+              label: `#${team.slug}`,
+            }),
+      };
+    })
+    .sort(({disabled: teamADisabled}, {disabled: teamBDisabled}) =>
+      teamADisabled === teamBDisabled ? 0 : teamADisabled ? 1 : -1
+    );
```

Comment:
Instead of sorting, can we actually have a new section in the dropdown? Kinda like what we did for the projects list